org.jacoco.core.test/src/org/jacoco/core/internal/analysis/MethodAnalyzerTest.java

```diff
@@ -514,6 +519,128 @@ public void jump_to_first_instruction_should_show_full_branch_coverage_when_prob
 		assertLine(1002, 0, 1, 0, 0);
 	}
 
+	// === Scenario: replace filtering of Kotlin safe call followed by elvis ===
+
+	/**
+	 * <pre>
+	 * data class B(val c: String)
+	 * fun safeCallElvis(b: B?): String =
+	 *     b?.c ?: ""
+	 * </pre>
+	 */
+	private void createKotlinSafeCallElvis() {
+		final Label l0 = new Label();
+		method.visitLabel(l0);
+		method.visitLineNumber(1001, l0);
+		final Label l1 = new Label();
+		final Label l2 = new Label();
+		method.visitVarInsn(Opcodes.ALOAD, 0);
+		method.visitInsn(Opcodes.DUP);
+		// probe[0] when jump to l1
+		method.visitJumpInsn(Opcodes.IFNULL, l1);
+		method.visitMethodInsn(Opcodes.INVOKEVIRTUAL, "B", "getC",
+				"()Ljava/lang/String;", false);
+		method.visitInsn(Opcodes.DUP);
+		// probe[1] when jump to l2
+		method.visitJumpInsn(Opcodes.IFNONNULL, l2);
+		// probe[2] is unreachable because `b.c` can not be null
+		method.visitLabel(l1);
+		method.visitInsn(Opcodes.POP);
+		method.visitLdcInsn("");
+		// probe[3]
+		method.visitLabel(l2);
+		// probe[4]
+		method.visitInsn(Opcodes.ARETURN);
```

Comment:
As a side note:

Every time I touch `MethodAnalyzerTest` I struggle to recall/realize how/where probes are inserted. Maybe it becomes easier after every touch, but still. That's why I started to add comments in the form `// probe[id]` - they at least help to recall, but unfortunately not all tests have them, and unfortunately they do not appear magically for new tests 😅 

IMO it would be nice to have a way to automatically show the places of probes or even better - assert them. Moreover `MethodAnalyzerTest` is already indirectly testing `LabelFlowAnalyzer` and `MethodProbesAdapter` and IMO contains more interesting (and far more) scenarios than their tests.